modules/foreman_provisioning_template.py

```diff
@@ -337,13 +346,15 @@ def main():
     state = template_dict.pop('state')
     file_name = template_dict.pop('file_name', None)
 
-    if file_name:
+    if file_name or 'template' in template_dict:
+        if file_name:
+            parsed_dict = parse_template_from_file(file_name, module)
+        else:
+            parsed_dict = parse_template(template_dict['template'], module)
+        # sanitize name from template data
+        if 'name' in parsed_dict and parsed_dict['name'] == '*':
+            module.fail_json(msg="Cannot use '*' as a template name!")
```

Comment:
If a template were to be named '*', which does not sound useful, the code would run through the bulk action branches.
If you want to name your template '***really important finish template***', which in turn might be useful, the module is not going to stop you.